### PR TITLE
VMware Plugin: Fix transformer issues

### PR DESCRIPTION
**Pull request by sduehr**
### Thank you for contributing to the Bareos Project!

This PR fixes multiple issue with the transformation of VM config data for recreation of VMs on restore.

Additionally, the vmware_cbt_tool.py can now dump the virtual machine config metadata to a JSON file, this can be helpful to debug this type of issue.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

File: docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/VMwarePlugin.rst.inc [hunk 1]
```diff
@@ -329,9 +330,18 @@ As of https://kb.vmware.com/s/article/2075984 manually enabling CBT is currently
                            disabled)
      --listall             List all VMs in the given datacenter with UUID and
                            containing folder
+     --sslverify           Force SSL certificate verification
+     --dumpvmconfig        Dump VM config metadata to JSON file
 
-Note: the options :command:`--vm-uuid` and :command:`--listall` have been added in version :sinceVersion:`17.2.8: VMware Plugin: new options in vmware\_cbt\_tool.py`, the tool is also able now to process non-ascii character arguments for the :command:`--folder` and :command:`--vmname` arguments and vApp names can be used like folder name components. With :command:`--listall` all VMs in the given datacenter are reported
-in a tabular output including instance UUID and containing Folder/vApp name.
+Note: the options :command:`--vm-uuid` and :command:`--listall` have been added in version :sinceVersion:`17.2.8: VMware Plugin: new options in vmware\_cbt\_tool.py`, the tool is also able now to process non-ascii character arguments for the :command:`--folder` and :command:`--vmname` arguments and vApp names can be used like folder name components.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/VMwarePlugin.rst.inc:336]: This note is not rendered as the other notes in the documentation , see this file line 287. Probably it makse sense to  format it like the other notes?

File: core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py [hunk 2]
```diff
@@ -3375,6 +3384,70 @@ def _transform_virtual_floppy(self, device):
...
+        if device["connectable"]:
+            add_device.connectable = self._transform_connectable(device)
+
+        # Looks like negative values must not be used for default devices,
+        # the VirtualPCIController always seems to have key 100.
+        add_device.controllerKey = device["controllerKey"]
+        if device["controllerKey"] != 100:
+            add_device.controllerKey = device["controllerKey"] * -1
+
+        add_device.unitNumber = device["unitNumber"]
```
Comment (pstorz) [inline core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py:3417]: This block is the same as line 3438 and following. Maybe that can be somehow unified?

**Review by pstorz — CHANGES_REQUESTED** [review]
See comments

File: docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/VMwarePlugin.rst.inc (hunk 1 above)
Comment (sduehr) [reply docs/manuals/source/TasksAndConcepts/Plugins/FileDaemonPlugins/VMwarePlugin.rst.inc]: Sure, it's formatted properly now.

File: core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py (hunk 2 above)
Comment (sduehr) [reply core/src/plugins/filed/python/vmware/BareosFdPluginVMware.py]: Good idea, it's unified now.

**Pull request merged**